### DISCUSSION TITLE
no function clause matching in :lists.usort/2

### DISCUSSION THREAD

**Discussion by DnOberon** [erlef/oidcc#279] [Q&A] [answered]
Hello it's me again!

Trying to create redirect URL gives me this error - using the config gleaned from this (if you can reach it https://identity-preview.inl.gov/.well-known/openid-configuration)

```
[error] GenServer #PID<0.528.0> terminating
** (FunctionClauseError) no function clause matching in :lists.usort/2
    (stdlib 5.0.2) lists.erl:1259: :lists.usort(#Function<1.25876173/2 in :oidcc_authorization.deprioritize_none_alg/1>, :undefined)
    (oidcc 3.0.1) src/oidcc_authorization.erl:233: :oidcc_authorization.attempt_request_object/2
    (oidcc 3.0.1) src/oidcc_authorization.erl:83: :oidcc_authorization.create_redirect_url/2

```
    
    I've verified my context is being pulled in correctly, and can give the  dbg output of that on request. 

```
    {:ok, context} =
      Oidcc.ClientContext.from_configuration_worker(
        Ingest.Application.OneID,
        config[:client_id],
        config[:client_secret]
      )

    dbg(
      Oidcc.Authorization.create_redirect_url(context, %{
        redirect_uri: "http://localhost:4000/users/log_in"
      })
    )
 ```

On the latest version.

**Comment by DnOberon** [comment]
I'm fairly certain something is misconfigured somewhere in the IdP but I'm not sure. Auth0 now works fine.

**Comment by maennchen** [comment, marked as answer]
Tracked as a bug in #280

**Reply by DnOberon** [reply]
Thanks!

**Reply by maennchen** [reply]
@DnOberon Bugfix released in `3.1.0-beta.1`. Can you test?

**Reply by DnOberon** [reply]
@maennchen that worked perfectly - thank you so much for looking into it!

**Comment by DnOberon** [comment]
Will do! I’ll let you know
